WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java

```diff
@@ -376,7 +386,40 @@ public void onRefreshRequested(StatsService.StatsEndpointsEnum[] endPointsNeedUp
         if (mIsUpdatingStats) {
             return;
         }
-        refreshStats(mCurrentTimeframe, mRequestedDate, true);
+
+        // fragments that needs an update require it at almost the same time. Keep a list of fragments
+        // that require the update and call the service with the right parameters.
+        synchronized (fragmentsRefreshListSynchObj) {
```

Comment:
All correct above. Unfortunately, there isn't an easy way to achieve the same goal without rewriting the StatsService ( StatsService comes from the past ). That's the reason we need to wait a little bit (500ms) and check what are the fragments that need an update.